lib/spack/spack/solver/asp.py

```diff
@@ -3597,16 +3717,8 @@ def external_spec_selected(self, node, idx):
 
     def depends_on(self, parent_node, dependency_node, type):
         dependency_spec = self._specs[dependency_node]
-        edges = self._specs[parent_node].edges_to_dependencies(name=dependency_spec.name)
-        edges = [x for x in edges if id(x.spec) == id(dependency_spec)]
         depflag = dt.flag_from_string(type)
-
-        if not edges:
-            self._specs[parent_node].add_dependency_edge(
-                self._specs[dependency_node], depflag=depflag, virtuals=()
-            )
-        else:
-            edges[0].update_deptypes(depflag=depflag)
+        self._specs[parent_node].add_dependency_edge(dependency_spec, depflag=depflag, virtuals=())
```

Comment:
I don't understand why this PR would mean that we can't do the same edge unification logic from before. Is the issue that the edge could be modified later for one of the edge-paths but not for another so we delay unification? It seems like this could lead to multiple edges between the same two specs.

Reply:
This is a simplification since spack.Spec.add_dependency edge already does this unification (spec.py L1828-1833)

Reply:
Ah yes, I should've grokked that since I'm pretty sure I made the change that allowed it.